src/Datagrid/Pager.php

```diff
@@ -36,13 +38,14 @@ public function computeNbResult()
             $countQuery->setParameters($this->getParameters());
         }
 
-        $countQuery->select(sprintf(
-            'count(DISTINCT %s.%s) as cnt',
-            current($countQuery->getRootAliases()),
-            current($this->getCountColumn())
-        ));
+        $query = $countQuery->getQuery();
+        $query->setHint(
+            CountWalker::HINT_DISTINCT,
+            $countQuery instanceof ProxyQuery ? $countQuery->isDistinct() : true
+        );
+        $this->appendTreeWalker($query, CountWalker::class);
 
-        return $countQuery->resetDQLPart('orderBy')->getQuery()->getSingleScalarResult();
```

Comment:
are we removing the reset of the orderBy?

Reply:
We copy query before manipulation 
``$countQuery = clone $this->getQuery();``